### PR TITLE
Feature/course track progress

### PR DESCRIPTION
**Pull request by DanielRios491**
## Description

The Student Track Progress feature introduces a mechanism to track each student's progress through course lessons. This solution captures video watch states, completed resources, and lesson completion per enrollment, ensuring accurate monitoring and reporting of student activity across the platform. The implementation tries to have clear domain boundaries, type safety, and enables extensibility for future enhancements.

## Main Changes

- Added the StudentTrackProgress entity, with its value objects (EnrollmentId, VideoProgress, ResourceCompleted).
- Introduced a dedicated DTO and mappers for safe data transfer through layers domain, and application.
- Created repository interfaces and implementations for persistent storage operations.
- Built a service layer to orchestrate business logic, including progress calculation and lesson/resource completion.
- Provided a RESTful controller and routing, fully integrated with dependency injection for testability and modularity.
- Updated unit tests to cover all new business logic, including error handling and edge cases.

image

image

## Reviewers

@LucasMezzaJalaUniversity
@JoseCaJala
@cclaurevju
@ArielMonroy392
@MauricioLSalles
@ElamCanoJala
@AndreCarpio
@DanielRios491
@eduSpinouza

### CODE REVIEW

File: src/contexts/LearningContext/presentation/http/routes/StudentTrackProgressRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import { container } from "tsyringe";
+import StudentTrackProgressController from "../controllers/StudentTrackProgressController ";
+
+const controller = container.resolve(StudentTrackProgressController);
+
+const studentTrackProgressRoutes = Router({ mergeParams: true });
+
+/**
+ * @openapi
+ * /student-track-progress/{trackId}:
```
Comment (ElamCanoJala) [inline src/contexts/LearningContext/presentation/http/routes/StudentTrackProgressRoutes.ts:11]: Here would be somehing like `/student-track-progress/:trackId` or why is like that?

File: src/contexts/LearningContext/presentation/http/routes/StudentTrackProgressRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+ *         description: Internal server error
+ */
+studentTrackProgressRoutes.get(
+  "/enrollment/:enrollmentId",
+  controller.getByEnrollment
+);
+
+/**
+ * @openapi
+ * /student-track-progress/{trackId}:
```
Comment (ElamCanoJala) [inline src/contexts/LearningContext/presentation/http/routes/StudentTrackProgressRoutes.ts:64]: Here maybe the same

**Review by ElamCanoJala — APPROVED** [review]

**Review by AndreCarpio — APPROVED** [review]

**Review by eduSpinouza — APPROVED** [review]
